src/main/scala/org/spurint/maven/plugins/mima/MiMaMojo.scala

```diff
@@ -219,29 +240,38 @@ class MiMaMojo extends AbstractMojo {
           getLog.warn(s"Unable to fetch previous artifact: $e")
           Option.empty
       }) match {
-        case Success(file) => file
-        case Failure(ex) => throw ex
-      }
+      case Success(file) => file
+      case Failure(ex) => throw ex
+    }
   }
 
   private def fetchLatestReleaseVersion(): Option[String] = {
     this.remoteRepositories.asScala.foldLeft(Option.empty[String])({
       case (None, repo) =>
+        val serverOpt = Option(settings.getServer(repo.getId))
+        getLog.info(s"Repo: ${repo.getUrl} server: ${serverOpt.map(_.getId)}")
+        val credsOpt = serverOpt.map( {server => (server.getUsername, server.getPassword)})
         val url = new URL(s"${repo.getUrl}/${this.project.getGroupId.replace(".", "/")}/${this.project.getArtifactId}/maven-metadata.xml")
-        makeHttpRequest(url).flatMap({ input =>
+        makeHttpRequest(url, credsOpt).flatMap({ input =>
           val xml = XML.load(input)
           (xml \ "versioning" \ "release").headOption.map(_.text.trim)
         })
       case (v @ Some(_), _) => v
     })
   }
 
-  private def makeHttpRequest(url: URL, retriesLeft: Int = 3): Option[InputStream] = {
+  private def makeHttpRequest(url: URL, credsOpt: Option[(String, String)], retriesLeft: Int = 3): Option[InputStream] = {
     try {
       val conn = url.openConnection() match {
         case huc: HttpURLConnection => huc
         case x => throw new AssertionError(s"${x.getClass.getName} should be a HttpURLConnection")
       }
+      credsOpt.foreach { case (username, password) =>
+        val userPass = s"$username:$password"
+        val basicAuth = s"Basic ${new String(Base64.getEncoder.encode(userPass.getBytes))}"
```

Comment:
Can be simplified; instead of `encode()` & `new String()` you can use `encodeToString()`.

Also, never call the zero-arg `String.getBytes()`; always use the variant that takes a `Charset` (which should be `StandardCharsets.UTF_8` in this case).